docs/source/development/data_storage.rst

```diff
@@ -0,0 +1,24 @@
+.. _core-development:data-storage:
```

Comment:
There was some discussion about whether we should better name this section "Internal Architecture" or similar, because otherwise the difference between this section and Topics section is not really clear.

Reply:
What about "Data storage internals"?
This section is specifically about data storage.

Reply:
Sorry, I meant that the whole "Core Development" section would be renamed to "Internal Architecture".

Reply:
> What about "Data storage internals"?
> This section is specifically about data storage.

What the section is intended for is really to document the design of AiiDA but from a core developer's perspective. This is for knowledge transfer to current and future aiida-core maintainers. So it does not only deal with data storage internals, but the internals of the engine and lots of other things. Design of the ORM, the split in backend/frontend, usage of SqlAlchemy for the `QueryBuilder` etc.

Reply:
@sphuber I think you misunderstood our conversation a bit here...
anyhow, just have another look when I'm finished